jupyterhub/app.py

```diff
@@ -365,12 +366,33 @@ def _log_format_default(self):
         """override default log format to include time"""
         return "%(color)s[%(levelname)1.1s %(asctime)s.%(msecs).03d %(name)s %(module)s:%(lineno)d]%(end_color)s %(message)s"
 
+    extra_log_file = Unicode("", config=True)
+    extra_log_handler = Instance(
+        klass=logging.Handler,
+        allow_none=True,
+        config=True,
```

Comment:
We could also remove the mutually-exclusive part by making it an `extra_log_handlers` list, and making `extra_log_file` just create a `FileHandler` and append it to the list.